### PR TITLE
event pkg: Created event pkg

### PR DESCRIPTION
**Pull request by evgenLevin**

### CODE REVIEW

File: pkg/events/eventList.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ListEvents returns Events inventory in the given namespace.
+func ListEvents(
```
Comment (kononovn) [inline pkg/events/eventList.go:13]: Re-name to List

File: pkg/events/events.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	k8sv1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EventBuilder provides struct for Event object which contains connection to cluster.
+type EventBuilder struct {
```
Comment (kononovn) [inline pkg/events/events.go:14]: Please re-name to Builder

File: pkg/events/events.go [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+	// nsName defines Event namespace.
+	nsName string
+	// nsName defines Event name.
+	eventName string
+	// errorMsg used in discovery function before sending api request to cluster.
+	errorMsg string
+}
+
+// NewEventBuilder creates new instance of EventBuilder.
+func NewEventBuilder(apiClient *clients.Settings, name, nsname string) *EventBuilder {
```
Comment (kononovn) [inline pkg/events/events.go:28]: Please rename to NewBuilder

**Review by kononovn — DISMISSED** [review]
/lgtm

File: pkg/events/events.go [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+	if !builder.Exists() {
+		return nil, fmt.Errorf("event object %s doesn't exist in namespace %s", name, nsname)
+	}
+
+	return &builder, nil
+}
+
+// Exists checks whether the given Event exists.
+func (builder *Builder) Exists() bool {
```
Comment (achuzhoy) [inline pkg/events/events.go:63]: no glog entry

File: pkg/events/events.go [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+	glog.V(100).Infof("Checking if Event %s exists", builder.Objects.Name)
+
+	var err error
+	builder.Objects, err = builder.Get()
+
+	return err == nil || !k8serrors.IsNotFound(err)
+}
+
+// Get returns Event object.
+func (builder *Builder) Get() (*k8sv1.Event, error) {
```
[the remark this answers is not in the document]
Comment (evgenLevin) [reply pkg/events/events.go:77]: added

File: pkg/events/eventList.go [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+		glog.V(100).Infof("Events 'nsname' parameter can not be empty")
+
+		return nil, fmt.Errorf("failed to list Events, 'nsname' parameter is empty")
+	}
+
+	logMessage := fmt.Sprintf("Listing Events in the namespace %s", nsname)
+	passedOptions := metaV1.ListOptions{}
+
+	if len(options) > 1 {
+		glog.V(100).Infof("'options' parameter must be empty or single-valued")
```
Comment (sebrandon1) [inline pkg/events/eventList.go:25]: Why accept a variadic parameter if you only use the first `option`?

Comment (evgenLevin) [reply pkg/events/eventList.go:25]: We want to maintain "options" as an optional argument that shouldn't exceed one.

**Review by cdvultur — APPROVED** [review]
/lgtm

**Comment by trewest** [comment]
Hey @evgenLevin since your pkg is named events you can re-name the file eventList.go to just list.go

File: pkg/events/events.go [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+	if valid, err := builder.validate(); !valid {
+		return nil, err
+	}
+
+	glog.V(100).Infof(
+		"Collecting Event object %s in namespace %s",
+		builder.Objects.Name, builder.Objects.Namespace)
+
+	event := &k8sv1.Event{}
+	err := builder.apiClient.Get(context.TODO(), goclient.ObjectKey{
```
Comment (trewest) [inline pkg/events/events.go:87]: It looks like you're missing the event namespace in this operation

File: pkg/events/events.go [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+	glog.V(100).Infof("Checking if Event %s exists", builder.Objects.Name)
+
+	var err error
+	builder.Objects, err = builder.Get()
+
+	return err == nil || !k8serrors.IsNotFound(err)
+}
+
+// Get returns Event object.
+func (builder *Builder) Get() (*k8sv1.Event, error) {
```
Comment (trewest) [inline pkg/events/events.go:77]: @evgenLevin I may be mistaken but I don't believe you need this method. I think you can just use the method `apiClient.Events(builder.Definition.Namespace).Get()`

**Review by trewest — CHANGES_REQUESTED** [review]
Couple of comments

File: pkg/events/events.go (hunk 7 above)
Comment (evgenLevin) [reply pkg/events/events.go:87]: renamed

File: pkg/events/events.go (hunk 8 above)
Comment (evgenLevin) [reply pkg/events/events.go:77]: removed

**Comment by evgenLevin** [comment]
@trewest, thank you for your useful comments. Could you please conduct one more review?

**Review by trewest — APPROVED** [review]
/lgtm

**Pull request merged**